crates/polars-io/src/pl_async.rs

```diff
@@ -317,6 +317,34 @@ impl RuntimeManager {
     {
         self.rt.spawn_blocking(f)
     }
+
+    /// Run a task on the rayon threadpool. To avoid deadlocks, if the current thread is already a
+    /// rayon thread, the task is executed on the current thread after tokio's `block_in_place` is
+    /// used to spawn another thread to poll futures.
+    pub async fn spawn_rayon<F, O>(&self, func: F) -> O
```

Comment:
Made this a function, but I think we won't be using it much as we go into the new streaming engine.